src/Compiler/pars.fsy

```diff
@@ -2721,6 +2721,16 @@ attrUnionCaseDecl:
             let mDecl = unionRangeWithXmlDoc xmlDoc mDecl
             Choice1Of2 (SynEnumCase ($1, $3, fst $5, xmlDoc, mDecl, trivia))) }
 
+  | opt_attributes opt_access unionCaseName EQUALS recover
```

Comment:
IIRC recover can be either an error, or EOF.
If it's an EOF in this branch, will this lead to an error in all situations?

In your tests, you always have a `()` at the end, so the EOF situation does not happen.